Expand documentation

underwoo opened the pull request:
**Description**
Migration of the FMS description web page to a markdown file in the repository.

underwoo committed:
Expand README

Copied FMS web pages from www.gfdl.noaa.gov into README file.

Corrrected a few errors in the INSTALL.md file

underwoo committed:
Merge branch 'master' into expand_documentation

bensonr commented on INSTALL.md:
```diff
@@ -14,16 +14,15 @@ The following external libraries are required when building libFMS
 
 ## MPI Support
 
 The default way to build libMPI is with MPI support.  We have found that using
```
Should this be "libFMS with MPI support"?

bensonr replied:
This is marked as resolved, but I still see "build libMPI with MPI support" in the text.

underwoo replied:
My bad.  I thought I fixed it with the other update.  I'll have it updated soon.

underwoo replied:
Update pushed.

bensonr commented on README.md:
```diff
@@ -1,10 +1,50 @@
+
+![FMS Framework](docs/images/FMS.gif)
+
+Modeling frameworks for the construction of coupled models, made from independent model components, are now prevalent across this field. This framework, FMS, was one of the first to be developed — since the advent of the Cray T3E in 1998 — and is still in use and development today, using new architectures and new algorithms.
```
I think it should be "in use and _under_ development"

bensonr commented on INSTALL.md:
```diff
@@ -14,16 +14,15 @@ The following external libraries are required when building libFMS
 general, offer the best results when building with MPI support.  If you decide
-to not use an MPI aware you should pass the include and library locations to
-the `configure` script through the `CPPFLAGS`, `FCFLAGS` and `LDFLAGS` variables.
+to not use an MPI aware compiler, you should pass the include and library locations to the build system.
```
May want to insert a hard return to limit the line length.

underwoo committed:
Minor corrections to INSTALL and README files

Add CMake section to INSTALL.md, that points to the CMAKE_INSTALL

Updated lint_fms workflow to new version of simple_lint

wrongkindofdoctor commented on README.md:
```diff
@@ -1,10 +1,67 @@
+
+Today’s climate models simulate highly complex systems. In response to
+increasingly complex models, the climate community has developed tools and
+methodologies to facilitate the modeling process, developing modeling frameworks
```
`, developing modeling frameworks
that provide infrastructure and tools to structure climate model codes and
facilitate many common tasks`--this sentence is redundant, and can be deleted

wrongkindofdoctor commented on README.md:
```diff
@@ -1,10 +1,67 @@
-efficient development, construction, execution, and scientific interpretation
-of atmospheric, oceanic, and climate system models.
+efficient development, construction, execution, and scientific interpretation of
+atmospheric, oceanic, and climate system models.  FMS comprises of the
```
FMS ~~comprises~~ consists of

wrongkindofdoctor commented on README.md:
```diff
@@ -1,10 +1,67 @@
+atmospheric, oceanic, and climate system models.  FMS comprises of the
+following:
+
+1. A software infrastructure for constructing and running atmospheric, oceanic,
```
The numbering needs to be fixed

bensonr replied:
The numbering is fine when you view it out of the markup editor.

underwoo replied:
Markdown will add the correct numbering when rendered.

wrongkindofdoctor commented on README.md:
```diff
@@ -1,10 +1,67 @@
+orchestration of the time stepping, makefiles, and simple sample run scripts.
+This infrastructure should largely insulate FMS users from machine-specific
+details.
+1. A standardization of the interfaces between various component models.
```
A standardization of the interfaces between various component models, including ~~S~~ software...

underwoo committed:
Small edits to README and INSTALL

underwoo committed:
More edits in README based on comments

underwoo committed:
Add diag_table documentation.

Pulled diag_table out of diag_table.F90 file.

underwoo committed:
Very minor edit to README

underwoo commented:
The changes have been made as requested.  Please review again.

I also added a `diag_table.md` file.  I pulled the information out of the diag_table.F90 file.
